bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-persistence.js

```diff
@@ -165,15 +184,21 @@ export default function defineOHBlocks_Persistence (f7, isGraalJs) {
   */
   javascriptGenerator['oh_persist_changed'] = function (block) {
     const itemName = javascriptGenerator.valueToCode(block, 'itemName', javascriptGenerator.ORDER_ATOMIC)
+    const inputType = block.getInput('itemName').connection.targetBlock().outputConnection.getCheck()[0]
```

Comment:
Why do you not use `blockGetCheckedInputType` here?

Reply:
I am not surprised you ask.  Because it doesn't work.  I did but there is a particular condition in 

> if (!thisBlock || !connectedBlock) return ''

which in these cases will return nothing.

Reply:
I don't understand why it shouldn't work, for me it works fine:

```javascript
    const inputType = blockGetCheckedInputType(block, 'itemName')
```

Reply:
Strangely, it does work with my unit tests and I don't know why...
I replaced it with blockGetCheckedInputType.